lib/webbluetooth/bindings.js

```
@@ -0,0 +1,267 @@
+
+};
+
+util.inherits(NobleBindings, events.EventEmitter);
+
+NobleBindings.prototype._onOpen = function() {
+  console.log('on -> open');
+};
+
+NobleBindings.prototype._onClose = function() {
+  console.log('on -> close');
+
+  this.emit('stateChange', 'poweredOff');
+};
+
+NobleBindings.prototype.startScanning = function(serviceUuids, allowDuplicates) {
+  var self = this;
+  console.log('startScanning', serviceUuids, allowDuplicates);
+
+  if(!Array.isArray(serviceUuids)){
+    serviceUuids = [serviceUuids];
+  }
+
+  ble.requestDevice({ filters: [{ services: serviceUuids.map(addDashes) }] })
+    .then(function(device){
+      console.log('scan finished', device);
+      if(device){
+
+        var address = device.instanceID;
+        var rssi;
+        //TODO use device.adData when api is ready
+        //rssi = device.adData.rssi;
+
+        self._peripherals[address] = {
+          uuid: address,
+          address: address,
+          advertisement: {}, //advertisement,
+          rssi: rssi,
+          device: device,
+          cachedServices: {}
+        };
+
+        self.emit('discover', device.instanceID, {localName:device.name, serviceUuids: serviceUuids}, device.rssi);
+
+      }
+    }, function(err){
+      console.log('err scanning', err);
+    });
+
+  this.emit('scanStart');
+};
+
+NobleBindings.prototype.stopScanning = function() {
+  this._startScanCommand = null;
+
+  //TODO: need web api completed for this to work'=
+  this.emit('scanStop');
+};
+
+NobleBindings.prototype.connect = function(deviceUuid) {
+  var self = this;
+  console.log('connect', deviceUuid);
+  var peripheral = this._peripherals[deviceUuid];
+
+  // Attempts to connect to remote GATT Server.
+  peripheral.device.connectGATT()
+    .then(function(gattServer){
+      peripheral.gattServer = gattServer;
+      console.log('peripheral connected', gattServer);
+      self.emit('connect', deviceUuid);
+    }, function(err){
+      console.log('err connecting', err);
```

You can bubble up the error: ``self.emit('connect', deviceUuid, error);``